src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer.UnitTests/Utilities/AbstractLanguageServerClientTests.cs

```diff
@@ -0,0 +1,150 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.CodeAnalysis.LanguageServer.HostWorkspace;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.CodeAnalysis.Testing;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.LanguageServer.Protocol;
+using Roslyn.Test.Utilities;
+using Roslyn.Utilities;
+using Xunit.Abstractions;
+using LSP = Roslyn.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.UnitTests;
+
+public abstract partial class AbstractLanguageServerClientTests(ITestOutputHelper testOutputHelper) : IDisposable
+{
+    protected TestOutputLogger TestOutputLogger => new(testOutputHelper);
+    protected TempRoot TempRoot => new();
+    protected TempDirectory ExtensionLogsDirectory => TempRoot.CreateDirectory();
+
+    public Task InitializeAsync()
+    {
+        return Task.CompletedTask;
+    }
+
+    public void Dispose()
+    {
+        TempRoot.Dispose();
+    }
+
+    private protected async Task<TestLspClient> CreateCSharpLanguageServerAsync(
+        [StringSyntax(PredefinedEmbeddedLanguageNames.CSharpTest)] string markupCode,
+        bool includeDevKitComponents,
+        bool debugLsp = false)
+    {
+        string code;
+        int? cursorPosition;
+        ImmutableDictionary<string, ImmutableArray<TextSpan>> spans;
+        TestFileMarkupParser.GetPositionAndSpans(markupCode, out code, out cursorPosition, out spans);
+
+        // Write project file
+        var projectDirectory = TempRoot.CreateDirectory();
+        var projectPath = Path.Combine(projectDirectory.Path, "Project.csproj");
+        await File.WriteAllTextAsync(projectPath, $"""
+            <Project Sdk="Microsoft.NET.Sdk">
+              <PropertyGroup>
+                <OutputType>Library</OutputType>
+                <TargetFramework>net{Environment.Version.Major}.0</TargetFramework>
```

Comment:
I'm a little hesitant to make this dynamic based on the runtime.  IMHO this should pull from the test project target framework if at all possible (or at least manually line it up)

Reply:
Updated to 8.0 (with comment) to match `$(NetVSCode)`